### PR TITLE
OCP4 migration

### PR DESCRIPTION
**Pull request by ShellyXueHan**
Changes including:
- updated based images for SSO and DB
- added nsp
- update BCDK pipeline dependencies and npm wrapper (wrapper needed otherwise missing tools)
- doc updates

### CODE REVIEW

File: .jenkins/docker/contrib/jenkins/configuration/scripts.groovy.d/on-gh-event.groovy [hunk 1]
```diff
@@ -36,74 +36,37 @@ static Map exec(List args, File workingDirectory=null, Appendable stdout=null, A
...
-                                    println exec(['oc', "--namespace=${reference[0]}", 'tag', "${reference[1]}", '--delete=true'])
-                                }
-                            })
-                        }
-
-                        //oc get dc -l 'env-id=pr-19,env-name!=prod' -o 'jsonpath={range .items[*]}{range .spec.triggers[*]}{.imageChangeParams.from.namespace}/{.imageChangeParams.from.name}{"\n"}{end}{end}'
-                        println exec(['oc', "--namespace=${namespace}", 'delete', 'all', '-l', selector])
-                        println exec(['oc', "--namespace=${namespace}", 'delete', 'PersistentVolumeClaim,Secret,ConfigMap,RoleBinding', '-l', selector])
-                    })
+                    println exec(['mkdir', '-p', gitWorkDir.getAbsolutePath()])
```
Comment (patricksimonian) [inline .jenkins/docker/contrib/jenkins/configuration/scripts.groovy.d/on-gh-event.groovy:54]: how did you figuire this out?

**Review by patricksimonian — APPROVED** [review]

File: .jenkins/docker/contrib/jenkins/configuration/scripts.groovy.d/on-gh-event.groovy (hunk 1 above)
Comment (ShellyXueHan) [reply .jenkins/docker/contrib/jenkins/configuration/scripts.groovy.d/on-gh-event.groovy:54]: ref: https://github.com/bcgov/RSBC-DataHub-API

File: .jenkins/.pipeline/lib/build.js [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+  objects.push(...oc.processDeploymentTemplate(`${templatesLocalBaseUrl}/build-master.yaml`, {
     'param':{
       'NAME': phases[phase].name,
       'SUFFIX': phases[phase].suffix,
       'VERSION': phases[phase].tag,
       'SOURCE_REPOSITORY_URL': oc.git.http_url,
-      'SOURCE_REPOSITORY_REF': oc.git.ref
+      'SOURCE_REPOSITORY_REF': oc.git.ref,
+      'SOURCE_IMAGE_STREAM_NAMESPACE': phases[phase].namespace,
+      'SOURCE_IMAGE_STREAM_TAG': 'jenkins-basic:prod'
```
Comment (arcshiftsolutions) [inline .jenkins/.pipeline/lib/build.js:22]: Should we externalize this value for the image tag?

**Review by arcshiftsolutions — APPROVED** [review]
Well done Shelly 👍

File: .jenkins/.pipeline/lib/build.js (hunk 2 above)
Comment (ShellyXueHan) [reply .jenkins/.pipeline/lib/build.js:22]: yea that's me being lazy thinking that we'll switch to another cicd tool :P let me update it!

**Pull request merged**